src/audio/kpb.c

```diff
@@ -786,17 +786,17 @@ static void kpb_init_draining(struct comp_data *kpb, struct kpb_client *cli)
 		trace_kpb("kpb_init_draining(), schedule draining task");
 
 		/* Add one-time draining task into the scheduler. */
-		kpb->draining_task_data.sink = kpb->cli_sink;
+		kpb->draining_task_data.sink = kpb->host_sink;
 		kpb->draining_task_data.history_buffer = buff;
 		kpb->draining_task_data.history_depth = history_depth;
 		kpb->draining_task_data.state = &kpb->state;
 
 		/* Pause selector copy. */
-		kpb->rt_sink->sink->state = COMP_STATE_PAUSED;
+		kpb->sel_sink->sink->state = COMP_STATE_PAUSED;
```

Comment:
Is this really needed? I thought we are returning `PPL_STATUS_PATH_STOP` on draining. I would really like to avoid setting component's state internally without kernel's knowledge. I guess detect pipeline should be stopped explicitly by the host on received notification? @lbetlej Any thoughts?

Reply:
I agree this setting here is redundant as well as kernel taking care of it. The only concern I have is timing. I mean when we get FW notification about detection we can stop further copy imediately inside FW. Kernel on the other hand first need to wait for IPC about notification, then send another one aimed to stop some pipeline.